docs/en/table_design/table_types/primary_key_table.md

```diff
@@ -1,159 +1,212 @@
+
+- In the CREATE TABLE statement, the primary key columns must be defined before other columns.
+- The primary key columns must include partitioning and bucketing columns.
+- The primary key columns support the following data types: numeric (including integers and BOOLEAN), string, and date (DATE and DATETIME).
+- The maximum length of a encoded primary key value is 128 bytes.
+- The primary key cannot be modified after table creation.
+- For data consistency purposes, the primary key values cannot be updated.
+
+### Primary key index
+
+The primary key index is used to store the mapping between the primary key values and the locations of the data rows identified by the primary key values. Typically, the primary key indexes of relevant tablets are loaded into memory only during data loading (which involves a batch of data changes). You can consider persisting the primary key indexes after comprehensively evaluating the performance requirement for queries and updates, as well as the memory and disk.
+
+<Tabs groupId="primary key Index">
+
+  <TabItem value="example1" label="Persistent primary keyIndex" default>
+
+When `enable_persistent_index` is set to `true` (default), the primary key indexes can be persisted to the disk. During loading, a small portion of the primary key indexes is loaded in memory, while the majority is stored on disk to avoid taking up too much memory. In general, query and update performance of the table with the persistent primary key indexes is nearly equivalent to that of the table with the fully in-memory primary key indexes.
+
+If the disk is a SSD, it is recommended to set it to true. If the disk is HDD and the load frequency is not high, you can also set it to true.
+
+:::note
 
-## Usage notes
+Since v3.1.4, Primary Key tables created in StarRocks shared-data clusters further support index persistence onto local disks.
 
-- Take note of the following points about the primary key of a table:
-  - The primary key is defined by using the `PRIMARY KEY` keyword.
+:::
 
-  - The primary key must be created on columns on which unique constraints are enforced, and the names of the primary key columns cannot be changed.
+</TabItem>
 
-  - The primary key columns can be any of the following data types: BOOLEAN, TINYINT, SMALLINT, INT, BIGINT, LARGEINT, STRING, VARCHAR, DATE, and DATETIME. However, the primary key columns cannot be defined as `NULL`.
+<TabItem value="example2" label="fully in-memory primary key index">
 
-  - The partition column and the bucket column must participate in the primary key.
+When `enable_persistent_index` is set to `false`, the primary key indexes are not persisted to the disk,  that is, the primary key indexes are fully stored in memory. During loading, the primary key indexes of tablets related to the data loaded will be loaded into memory, which may result in higher memory consumption. (If a tablet has not had data loaded for a long time, its primary key index will be released from memory).
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/108eaf76-14d5-4e9f-8afa-8e4553564b0e)
Put the period (.) inside the parentheses.